### DISCUSSION TITLE
@message_handler or @channel_post_handler for a group with topics?

### DISCUSSION THREAD

**Discussion by ZETECdev** [eternnoir/pyTelegramBotAPI#1891] [Q&A]
Hi... i dont find a solution

i need handler all messages of a specific topic of a group. Api v4.6 allow message_thread_id for reference to the different topics of a group

Bu i dont find how to handle messages from  a specific topic... 

Someone knows? Thx

**Comment by Badiboy** [comment]
Check the message_thread_id.

![image](https://user-images.githubusercontent.com/5613421/214557416-7309c22d-f5a0-4505-ba0b-8dc9c454e02a.png)

https://core.telegram.org/bots/api#message

**Comment by ZETECdev** [comment]
for send_message method works fine but for handlers i dont know how work

@bot.message_handler(content_types=['text', 'photo'], chat_id=config.BBMCHATID, message_thread_id=310406)

Here i not receved errors, but dosnt work, this is correct?


very thanks

**Comment by Badiboy** [comment]
No. Use lambda.
